sync/src/main/java/tech/pegasys/artemis/sync/AttestationManager.java

```diff
@@ -100,6 +100,11 @@ private void onBlockImported(final ImportedBlockEvent blockImportedEvent) {
               pendingAttestations.remove(attestation);
               processAttestation(attestation);
             });
+    block
+        .getMessage()
+        .getBody()
+        .getAttestations()
+        .forEach(attestationProcessor::processAttestation);
```

Comment:
If we're importing old blocks I believe this will wind up failing the check in `ForkChoiceUtil.on_attestation`:
```
if (!target.getEpoch().equals(previous_epoch) && !target.getEpoch().equals(current_epoch)) {
      return AttestationProcessingResult.invalid(
          "on_attestation: Attestations must be from the current or previous epoch");
    }
```
because it's computing the current epoch based on the store, which is always at the current time, even if we haven't sync'd all blocks yet. So I think this winds up causing us to fail to import any blocks via sync...

Reply:
Yeah makes sense. I'm wondering if we shouldn't run on_attestation on any attestation that is from syncing blocks.

Reply:
The spec says we should, but it would fail validation because they're too old.  It's unclear exactly what we should do about that.

Reply:
And we do need to ensure that checkpoint states are stored even for old blocks.  Latest messages may not matter though.